resources/agent.rb

```diff
@@ -35,32 +35,66 @@
                                    "backend-url": ['ws://127.0.0.1:8081'],
                                  }
 action :install do
-  packagecloud_repo new_resource.repo do
-    type value_for_platform_family(
-      %w(rhel fedora amazon) => 'rpm',
-      'default' => 'deb'
-    )
-  end
+  # Linux installation - source package
+  if node['platform'] != 'windows'
+    packagecloud_repo new_resource.repo do
+      type value_for_platform_family(
+        %w(rhel fedora amazon) => 'rpm',
+        'default' => 'deb'
+      )
+    end
 
-  package 'sensu-go-agent' do
-    if latest_version?(new_resource.version)
-      action :upgrade
-    else
-      version new_resource.version
-      action :install
+    # Installs/upgrades sensu-go-agent package
+    package 'sensu-go-agent' do
+      if latest_version?(new_resource.version)
+        action :upgrade
+      else
+        version new_resource.version
+        action :install
+      end
     end
-  end
 
-  # render template at /etc/sensu/agent.yml
-  file ::File.join(new_resource.config_home, 'agent.yml') do
-    content(stringify_keys(new_resource.config))
+    # render template at /etc/sensu/agent.yml for linux
+    file ::File.join(new_resource.config_home, 'agent.yml') do
+      content(stringify_keys(new_resource.config))
+    end
+
+    # Enable and start the sensu-agent service
+    service 'sensu-agent' do
+      if node['platform'] == 'ubuntu' && node['platform_version'].to_f == 14.04
+        provider Chef::Provider::Service::Init
+        action :start
+      else
+        action [:enable, :start]
+      end
+    end
   end
 
-  service 'sensu-agent' do
-    if node['platform'] == 'ubuntu' && node['platform_version'].to_f == 14.04
-      provider Chef::Provider::Service::Init
-      action :start
-    else
+  # Installs msi for Sensu-Go
+  if node['platform'] == 'windows'
+    windows_package 'sensu-go-agent' do
+      source 'https://s3-us-west-2.amazonaws.com/sensu.io/sensu-go/5.10.0/sensu-go-agent_5.10.0.4171_en-US.x64.msi'
```

Comment:
I would think we want to expose this as an attribute, thats fine to have a sane default but we should not be hardcoding versions or URls here. Given how wildly different the versioning is I would say something like `windows_msi_source` and `msi_version` or something along those lines would be appropriate.

Reply:
What would you suggest pinning the default to as we lack a `:latest` due to the current pipeline for windows packaging. 

Sensu.io says they are working on solving this, but have not mentioned a timeline.

Reply:
agreed we should pin and those are fine defaults for now let's  just do it in a way that allows people to override without having to use `edit_resource`.